### ISSUE TITLE
[0.63] Alert hangs or may not even show up only in Release Builds

### ISSUE THREAD

**Issue by creambyemute** [microsoft/react-native-windows#8033]
## Environment
Run the following in your terminal and copy the results here.
1. `npx react-native --version`: 5.0.1
2. `npx react-native info`: 
```
System:
    OS: Windows 10 10.0.19042
    CPU: (12) x64 AMD Ryzen 5 5600X 6-Core Processor
    Memory: 17.05 GB / 31.95 GB
  Binaries:
    Node: 14.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.11 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 29.0.0, 29.0.1, 29.0.2, 29.0.3, 30.0.0, 30.0.1, 30.0.2, 30.0.3
      System Images: android-28 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Version  4.1.0.0 AI-201.8743.12.41.7042882
    Visual Studio: 16.9.31313.79 (Visual Studio Community 2019)
  Languages:
    Java: 1.8.0_282 - C:\Program Files\OpenJDK\openjdk-8u282-b08\bin\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.2 => 0.64.2
    react-native-windows: ^0.64.0-0 => 0.64.11
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Create a new 0.63 App
2. `yarn add react-native-alert-async` or create a promise alert yourself.
3. Replace App.js with: 
```javascript
const App = () => {
  const [text, setText] = React.useState<String>('Nothing');

  const showAlertOne = async () => {
    return await AlertAsync(
      'Alert Async Test1 Title',
      'Alert Async Test1 Message',
      [
        {
          text: 'OK',
          onPress: () => Promise.resolve(true),
        },
        {
          text: 'Cancel',
          onPress: () => Promise.resolve(false),
        },
      ],
      {cancelable: false},
    );
  };

  const showAlertTwo = async () => {
    return await AlertAsync(
      'Alert Async Test2 Title',
      'Alert Async Test2 Message',
      [
        {
          text: 'OK',
          onPress: () => Promise.resolve(true),
        },
        {
          text: 'Cancel',
          onPress: () => Promise.resolve(false),
        },
      ],
      {cancelable: false},
    );
  };

  const onPressShowAlertDialogAndUpdateState = async event => {
    const alertOneResult = await showAlertOne();
    setText(`alertOneResult: ${alertOneResult}`);
    if (alertOneResult) {
      const alertTwoResult = await showAlertTwo();
      setText(`alertTwoResult: ${alertTwoResult}`);
    }
  };

  return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView>
        <ScrollView
          contentInsetAdjustmentBehavior="automatic"
          style={styles.scrollView}>
          <View style={styles.body}>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>Alert State:</Text>
              <Text>{text}</Text>
              <TouchableOpacity onPress={onPressShowAlertDialogAndUpdateState}>
                <Text>Press to show Alerts</Text>
              </TouchableOpacity>
            </View>
          </View>
        </ScrollView>
      </SafeAreaView>
    </>
  );
};
```

## Expected Results
Expected: State updates to happen, second alert to show up fast after answering first alert with OK.

Actually happening: State updates are either not reflected or very slow to happen. Second alert sometimes doesn't show or needs an interaction to show up or just takes several seconds (either of those 3 scenarios can happen)

## Snack, code example, screenshot, or link to a repository:
See repro steps + code example above

## Additional Information / Question
I have tried the same in a fresh 0.64 App and there it works as expected 👍.  

Do you by chance know what might cause this in 0.63? I'm atm still locked into 0.63 and wondered if there is an easy patch/solution to this?

**Comment by chrisglein** [comment]
> Do you by chance know what might cause this in 0.63? I'm atm still locked into 0.63 and wondered if there is an easy patch/solution to this?

There was a big threading behavior change that could have been what affected the behavior here. There were also some JSI event bug fixes that could be relevant, given this only happening in Release builds for you.

Step to valid here would be to look at this in RNTester on the 0.63 stable branch.

**Comment by creambyemute** [comment]
I totally did not expect this to be fixed/backported to 0.63 anymore now that 0.65 is being released soon. So that's fine.

And I think you are right with the threading changes. I suspect the culprit to be the `jsidispatcher.post()` call
